### ISSUE TITLE
Integration for Lazy Minting Video Data as NFTs on IPFS with meToken Pricing

### ISSUE THREAD

**Issue by sirgawain0x** [creativeplatform/crtv3#72]
We would like to integrate a new feature where each video upload has the option to lazy mint the video data from IPFS as an NFT. The claim price for the video NFT can be set via the uploader's chosen meToken price available on Optimism or Base.

Feature Details:
- [ ] Lazy Minting: Enable lazy minting of video data on IPFS as NFTs.

- [ ] Users should have the option to mint their video data as NFTs during the upload process.

- [ ] Pricing Mechanism: Allow uploaders to set the claim price for their video NFTs using their chosen meToken.

- [ ] Support for meToken pricing should be available on both Optimism and Base.

Technical Requirements:

-     IPFS Integration: Ensure seamless integration with IPFS for storing video data.

-     NFT Minting: Implement lazy minting functionality for video NFTs.

-     meToken Support: Support meToken pricing for NFTs on Optimism and Base.

-     Blockchain Interaction: Ensure interactions with Optimism and Base blockchains for setting and claiming NFT prices.

Additional Context:
This feature will enhance the platform by providing users with the ability to monetize their video content through NFT minting and pricing flexibility.

**Comment by amochuko** [comment]
Am going to attempt this.

**Comment by amochuko** [comment]
Some questions please; 

1. Do I need a membership NFT for development?
2. Is the `upload` function available?

**Comment by sirgawain0x** [comment]
> Some questions please; 
> 
> 1. Do I need a membership NFT for development?

Yes. Sorry about that, I'll send that over to you now. How did you realize you needed a membership NFT?

**Comment by amochuko** [comment]
> > Some questions please;
> > 
> > 1. Do I need a membership NFT for development?
> 
> Yes. Sorry about that, I'll send that over to you now. How did you realize you needed a membership NFT?

Well, I was on the profile page and saw the message.

**Comment by liberalterian** [comment]
I'm not sure if this overlaps but I implemented the Thirdweb ERC1155 TS SDK to the upload process to lazy mint NFTs so that I could move forward with testing the video token gating webhook I setup. See my [recent comment on issue #77](https://github.com/creativeplatform/crtv3/issues/77#issuecomment-2544027702) for more information. 

I didn't realize this might overlap so I apologize if that is the case. I didn't do anything for the pricing but I did add a checkbox to the CreateInfo component to indicate that the user wants to token gate their new video.

Let me know what you think.

**Comment by amochuko** [comment]
I don't think it would. Carry on.
